pkg/clusterlogging/clusterlogging.go

```diff
@@ -0,0 +1,190 @@
+
+		builder.errorMsg = "The clusterLogging 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// Pull returns clusterLogging object if found.
+func (builder *ClusterLoggingBuilder) Pull() (*clov1.ClusterLogging, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting clusterLogging %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	clusterLogging := &clov1.ClusterLogging{}
+	err := builder.apiClient.Get(context.Background(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, clusterLogging)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return clusterLogging, err
+}
+
+// Discover fetches existing clusterLogging from cluster.
+func (builder *ClusterLoggingBuilder) Discover() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Pulling existing clusterLogging with name %s under namespace %s from cluster",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	clo := &clov1.ClusterLogging{}
+	err := builder.apiClient.Get(context.Background(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, clo)
+
+	builder.Object = clo
+
+	return err
+}
+
+// Create makes a clusterLogging in the cluster and stores the created object in struct.
+func (builder *ClusterLoggingBuilder) Create() (*ClusterLoggingBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the clusterLogging %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes clusterLogging from a cluster.
+func (builder *ClusterLoggingBuilder) Delete() (*ClusterLoggingBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the clusterLogging %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("clusterLogging cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.Background(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete clusterLogging: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// Exists checks whether the given clusterLogging exists.
+func (builder *ClusterLoggingBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if clusterLogging %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Pull()
```

Comment:
Use builder.Get() function here

Reply:
in you presentation two weeks ago it was Pull as a must in a list of methods. Do we start it again?
what is a final agreement with you and Trey: Pull or Get? or both, or maybe Discovery and Get?? or any other combination? I don't really follow your last decisions made...